### ISSUE TITLE
Lock Issues (Still)

### ISSUE THREAD

**Issue by OdinVex** [qualiaa/acpi-brightness#1]
You still have lock issues. I find the script executing for hours at a time constantly setting the brightness (which is fine) but it never closes and if I attempt to control brightness through another software even for just a moment, acpi-brightness will change it back. (It is in a loop setting brightness but never exiting?).

**Comment by qualiaa** [comment]
Yes, I noticed this, I haven't been bothered enough to fix it, but now I know someone else is using it I'll have a look :) 

I'm using pretty raw locking mechanisms at the moment (`mkdir`) but I think the implementation will be much more comprehensible and robust with `sem`.

I'll add a check that the current value is the value that we last set, and terminate if not. Hopefully that will fix your problem with other software.

**Comment by OdinVex** [comment]
I've been using it for quite a long time (well over a year?), though lately I've become more obsessive about fixing the small things with my installs.

**Comment by OdinVex** [comment]
I've been using it for quite a long time (well over a year?), though lately I've become more obsessive about fixing the small things with my installations.

*Edit* I forgot to follow-up about debugging this issue. I checked my redshift-hooks.log and found a number of errors relating to the execution of the script:

`Unrecognised: daytime` is repeated, so I've customized the initial brightness.sh with poor-man's debugging to print things out whenever those errors are printed from it.

`find: '/tmp/runtime-root': Permission denied`
`find: '/tmp/systemd-private-...service-CgiEWe': Permission denied`
`grep: /tmp/brightness.YOse: No such file or directory`
`/usr/bin/acpi-brightness: line 54: echo: write error: Invalid argument`

I also use `readonly backlight=/sys/class/backlight/amdgpu_bl0` as I use an AMD setup. I've also silenced the `find` errors with 2>/dev/null on line 100 (`local lockfiles=$(find "${TMPDIR:-/tmp}" -name "${lockfile_name}.*" 2>/dev/null)`).

I'm wondering if the long run-time is due to maybe some math issues relating to fade/transition. Not only that, but I recall the cmdline being the same every time I've had to terminate the process.

**Comment by OdinVex** [comment]
Redshift seems to have a backlight brightness option now.

*Edit* The redshift backlight option seems to be a fake gamma thing...it was horrible on my setups.
